engine/components/nationalinsurancenumberfield.js

```diff
@@ -0,0 +1,35 @@
+const { FormComponent } = require('.')
+const helpers = require('./helpers')
+
+const PATTERN = '^(?!BG|GB|NK|KN|TN|NT|ZZ)[ABCEGHJ-PRSTW-Z]\\s*[ABCEGHJ-NPRSTW-Z]\\s*\\d{1}\\s*\\d{1}\\s*\\d{1}\\s*\\d{1}\\s*\\d{1}\\s*\\d{1}\\s*[A-D]$'
```

Comment:
I think your plan was to strip whitespace client side, in which case this can be simplified as it was in RIL?

Reply:
my 2p, I think ws strip should happen on server just in case someone has js turned off

Reply:
Makes sense

Reply:
defo agree, in fact, we should do both. Let me take a look at the post handler again in the runner. I talked with Andy about implementing a method where each component knew how to 'post-process', 'clean-up', 'tidy' it's own input, but the problem is that the POST handler is generic so would fire for every component, and also on every page transition (on every POST). Thinking about this. Is there a separate handler for the final submission of a form? Maybe this is the correct location for invoking any code that is responsible for a final transform before the data captured in the form is sent on to the back office.